x-pack/plugins/siem/server/lib/kpi_hosts/query_general.dsl.ts

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { createQueryFilterClauses } from '../../utils/build_query';
+import { RequestBasicOptions } from '../framework';
+
+import { KpiHostsESMSearchBody } from './types';
+
+export const buildGeneralQuery = ({
+  filterQuery,
+  timerange: { from, to },
+  sourceConfiguration: {
+    fields: { timestamp },
+    logAlias,
+    auditbeatAlias,
+    packetbeatAlias,
+    winlogbeatAlias,
+  },
+}: RequestBasicOptions): KpiHostsESMSearchBody[] => {
+  const filter = [
+    ...createQueryFilterClauses(filterQuery),
+    {
+      range: {
+        [timestamp]: {
+          gte: from,
+          lte: to,
+        },
+      },
+    },
+  ];
+
+  const dslQuery = [
+    {
+      index: [logAlias, auditbeatAlias, packetbeatAlias, winlogbeatAlias],
+      allowNoIndices: true,
+      ignoreUnavailable: true,
+    },
+    {
+      aggregations: {
+        hosts: {
+          cardinality: {
+            field: 'host.name',
+          },
+        },
+        hosts_histogram: {
+          auto_date_histogram: {
```

Comment:
Sometimes this is not giving me the bucket number I request for, is this the right way to ask for data?
<img width="1680" alt="Screenshot 2019-05-06 at 22 05 54" src="https://user-images.githubusercontent.com/6295984/57232708-bef58700-704f-11e9-8a29-1267245fb503.png">
<img width="1680" alt="Screenshot 2019-05-06 at 22 06 48" src="https://user-images.githubusercontent.com/6295984/57232709-bef58700-704f-11e9-85d7-5e6dd19b7389.png">

Reply:
See my above comment -- I think you should be good here.